### ISSUE TITLE
Setting isActive for URL type in menu's doesn't work

### ISSUE THREAD

**Issue by ghost** [rainlab/pages-plugin#126]
I noticed that for my menu item with type URL the active class wasn't set.
Discovered that in `classes/Menu.php` the `Request::path()` is converted to a full url with `URL::to()` and later compared to a URL that is not. 

Below you can see the code from line 136. You can see at the end that the `$currentUrl` is compared to the `$item->url` which is only a path.

``` PHP
$currentUrl = Request::path();

if (!strlen($currentUrl)) {
    $currentUrl = '/';
}

$currentUrl = Str::lower(URL::to($currentUrl));

$activeMenuItem = $page->activeMenuItem ?: false;
$iterator = function($items) use ($currentUrl, &$iterator, $activeMenuItem) {
    $result = [];

    foreach ($items as $item) {
        $parentReference = new MenuItemReference;
        $parentReference->title = $item->title;
        $parentReference->code = $item->code;
        $parentReference->viewBag = $item->viewBag;

        /*
         * If the item type is URL, assign the reference the item's URL and compare the current URL with the item URL
         * to determine whether the item is active.
         */
        if ($item->type == 'url') {
            $parentReference->url = $item->url;
            $parentReference->isActive = $currentUrl == Str::lower($item->url) || $activeMenuItem === $item->code;
        }
        else {
            /*
             * If the item type is not URL, use the API to request the item type's provider to
             * return the item URL, subitems and determine whether the item is active.
             */
            $apiResult = Event::fire('pages.menuitem.resolveItem', [$item->type, $item, $currentUrl, $this->theme]);
```

I believe the converted `$currentUrl` is only used later on in the else statement for firing the `pages.menuitem.resolveItem` event. It would be more logical to convert it here right?
This would result in following code;

``` PHP
$currentUrl = Str::lower(Request::path());

if (!strlen($currentUrl)) {
    $currentUrl = '/';
}

$activeMenuItem = $page->activeMenuItem ?: false;
$iterator = function($items) use ($currentUrl, &$iterator, $activeMenuItem) {
    $result = [];

    foreach ($items as $item) {
        $parentReference = new MenuItemReference;
        $parentReference->title = $item->title;
        $parentReference->code = $item->code;
        $parentReference->viewBag = $item->viewBag;

        /*
         * If the item type is URL, assign the reference the item's URL and compare the current URL with the item URL
         * to determine whether the item is active.
         */
        if ($item->type == 'url') {
            $parentReference->url = $item->url;
            $parentReference->isActive = $currentUrl == Str::lower($item->url) || $activeMenuItem === $item->code;
        }
        else {
            /*
             * If the item type is not URL, use the API to request the item type's provider to
             * return the item URL, subitems and determine whether the item is active.
             */
            $apiResult = Event::fire('pages.menuitem.resolveItem', [$item->type, $item, URL::to($currentUrl)

, $this->theme]);
```

**Comment by daftspunk** [comment]
If this is a bug can you provide step by step replication instructions, along with expected vs actual results.

**Comment by bijsterz** [comment]
I've created a menu option for the home page, type is URL. the entered url is "/"
I've also created a menu option to a CMS page and a menu option for all static pages.

Using the following `items.htm` partial to display my menu (copied mostly from the original);

```PHP
{% for item in items if not item.viewBag.isHidden %}
    <li role="presentation" class="{{ item.isActive ? 'active' : '' }} {{ item.isChildActive ? 'child-active' : '' }} {{ item.viewBag.cssClass }}">
        {% if item.url %}
            <a href="{{ item.url }}" {{ item.viewBag.isExternal ? 'target="_blank"' }}>
                {{ item.title }}
            </a>
        {% else %}
            <span>{{ item.title }}</span>
        {% endif %}

        {% if item.items %}
            <ul>{% partial __SELF__ ~ "::items" items=item.items %}</ul>
        {% endif %}
    </li>
{% endfor %}
```

Expected result is that the active class gets set for URL type menu items.
Actual result is that is is not set for URL type menu items.

**Comment by bijsterz** [comment]
I've created a pull request. I'm a bit new to GitHub.

**Comment by TomaszN007** [comment]
Ty for that fix! I thought that I was doing something wrong in the partial code but the problem was in this class :(

**Comment by LaravelNick** [comment]
Could someone please kindly explain what I need to do to work around this bug, as I'm experiencing exactly the same issue as described above, using the latest build of October CMS ?

**Comment by bijsterz** [comment]
Hi LaravelNick.

To fix this you need to open the file `plugins/rainlab/pages/classes/Menu.php` and edit the following lines;

Change following line
`130 $currentUrl = Request::path();`
To
`130 $currentUrl = Str::lower(Request::path());`

Remove following line
`136 $currentUrl = Str::lower(URL::to($currentUrl));`

Change following line
`159 $apiResult = Event::fire('pages.menuitem.resolveItem', [$item->type, $item, $currentUrl, $this->theme]);`
To
`159 $apiResult = Event::fire('pages.menuitem.resolveItem', [$item->type, $item, URL::to($currentUrl), $this->theme]);`

Good luck ;)

**Comment by LaravelNick** [comment]
Hi bijsterz,

Thank you so much for that, it's really kind of you to post that and very helpful to me.

I will give it a try.

Many thanks :-)

**Comment by LaravelNick** [comment]
This bug is still present within the latest version of the plugin.

The solution that bijsterz kindly posted resolves the issue for single level URLs (e.g. /about-us) but not for multi level URLs (e.g. /about-us/our-team).

**Comment by LaravelNick** [comment]
It's a shame that this bug has still not been fixed in the Stable release of October CMS.

**Comment by JumpLink** [comment]
please fix

**Comment by daslicht** [comment]
Any news on this ?

**Comment by LukeTowers** [comment]
@bijsterz Please make your pull request to this repository, not your own. @daslicht We'll take a look at it once we can confirm the issue on our end and confirm the fix in the PR that will be made by @bijsterz.

**Comment by pleswi** [comment]
Please fix this bug. More than year without any action :(

**Comment by LukeTowers** [comment]
@pleswi make a PR to fix the issue.
The support demand for October is increasing, while the available time the founders have to give remains the same. These issues will be handled in time, however there are ways you can help:

- Submit a pull request that introduces this feature and wait for it to be reviewed
- [Make a financial contribution to the project](http://octobercms.com/fundraising) so we can allocate more time towards making improvements like these that benefit everyone

**Comment by bennothommo** [comment]
Fixed as of bad169125b4f5180e7e2a0253103e87e9abf30d7.
